app/generators/concerns/prefillable.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Gives access to handy methods to determine past responses of a measurement
+module Prefillable
+  extend ActiveSupport::Concern
+
+  # Finds the last response for the response's user and measurement
+  # @note the response found may not be part of the same protocol as the given response, but it will be of the same
+  # type of questionnaire. This is exactly the behaviour needed by UMO, where a user will at most respond once to a
+  # protocol, but all protocols have the same 7 initial measurements
+  # @return nil if the measurement doesn't need prefilling or no previous response exists, {#Response} otherwise
+  def previous_response(response)
+    return nil unless response&.needs_prefilling?
+
+    Response.joins(:measurement)
+            .where(
+              filled_out_by_id: response.person.id,
+              measurement: { questionnaire_id: response.measurement.questionnaire_id }
+            ).completed
+            .order(completed_at: :desc)
+            .first
+  end
+
+  def previous_response_content(response)
+    prev_response = previous_response(response)
+    return nil unless prev_response.present?
+
+    prev_response.remote_content&.content
+  end
+end
```

Comment:
So in relation to this, Frank some time ago added code to use a previous response's answers to use in the text of a questionnaire question, i.e., it would be used in a context saying "Previously, you answered the following question with the text: {{previous_v14}}." The questions would be annotated with a `uses: ` clause that loaded a previous response's values up for use in the question texts.

Now this differs from your case, because in Frank's case, we're replacing text in a question with text from a response, whereas in your case, we're prefilling a questionnaire with values from a previous response.

My only concern is that the code in `app/logic/previous_response_finder.rb` shares a lot of logic with `app/generators/concerns/prefillable.rb`. Essentially it's code duplication, although I believe your version to be better (with the nice SQL query and such). Is there a way you can change `app/logic/previous_reponse_finder.rb` to use your prefillable concern instead of duplicating the same logic?

Reply:
wow, the overlap is uncanny. I didn't know something like this existed. I'll try to merge it somehow

Reply:
k, came up with something